repair/task_manager_module.hh

```diff
@@ -48,9 +48,10 @@ private:
     std::vector<sstring> _data_centers;
     std::unordered_set<gms::inet_address> _ignore_nodes;
     bool _small_table_optimization;
+    bool _mixed_shard_optimization;
     std::optional<int> _ranges_parallelism;
 public:
-    user_requested_repair_task_impl(tasks::task_manager::module_ptr module, repair_uniq_id id, std::string keyspace, std::string entity, lw_shared_ptr<locator::global_vnode_effective_replication_map> germs, std::vector<sstring> cfs, dht::token_range_vector ranges, std::vector<sstring> hosts, std::vector<sstring> data_centers, std::unordered_set<gms::inet_address> ignore_nodes, bool small_table_optimization, std::optional<int> ranges_parallelism) noexcept
+    user_requested_repair_task_impl(tasks::task_manager::module_ptr module, repair_uniq_id id, std::string keyspace, std::string entity, lw_shared_ptr<locator::global_vnode_effective_replication_map> germs, std::vector<sstring> cfs, dht::token_range_vector ranges, std::vector<sstring> hosts, std::vector<sstring> data_centers, std::unordered_set<gms::inet_address> ignore_nodes, bool small_table_optimization, bool mixed_shard_optimization, std::optional<int> ranges_parallelism) noexcept
```

Comment:
Is it ok that RBNO will never use it?

Reply:
Very good point, we need to expose this to RBNO, it is especially important for replace, which doesn't work well with streaming.

Reply:
We will start with normal repair only and enable it for RBNO later.